application/models/thread.go

```diff
@@ -202,7 +202,9 @@ func (t *Thread) Load(fields ...string) error {
 	return nil
 }
 
-func (t *Thread) UnreadMessageCount(lastViewedAt time.Time) (int, error) {
+// UnreadMessageCount returns the number of messages on this thread that the current
+//  user has not created and for which the CreatedAt value is after the lastViewedAt value
+func (t *Thread) UnreadMessageCount(user *User, lastViewedAt time.Time) (int, error) {
```

Comment:
I don't like that `user` is a pointer. It indicates that the function could modify it. If it's for performance reasons, I suspect that optimizations eliminate any differences.

Reply:
Sounds good.